python_modules/dagster/dagster_tests/cli_tests/command_tests/test_materialize_command.py

```diff
@@ -118,3 +118,14 @@ def test_conflicting_partitions():
 def test_failure():
     result = invoke_materialize("fail_asset")
     assert result.exit_code == 1
+
+def test_run__cli_config_json():
+    with instance_for_test() as instance:
+        runner = CliRunner()
+
+        result = runner.invoke(asset_materialize_command, ["--select", "asset_with_config", "--cli-config-json", "{\"some_prop\": \"foo\"}"])
+
+        assert "RUN_SUCCESS" in result.output
+        assert instance.get_latest_materialization_event(AssetKey("asset_with_config")) is not None
+        assert result.exit_code == 0
+        # TODO: how to check the value of the materialized asset?
```

Comment:
@OwenKephart any guidance on how to assert the result of my new asset contains the `"foo"` string I passed?

Reply:
one option would be to do a context.log(f"{config.some_prop}_{config.some_prop}") within the body of the asset, which should show up in the result logs here. (doing the foo_foo sort of "confirms" that it's coming from the body of the op rather than somewhere else)

Reply:
And then check like this?
```py
assert "some_prop:foo" in result.output
```

Or do I need to look at the events?

Reply:
I've updated the code, would appreciate it if you could allow the tests to run in CI as I am having trouble with local environment.

Reply:
I was thinking:

```python
assert "foo_foo" in result.output
```

result.output in this case is just the raw string text produced by running that command, so no need to do any dagster-specific event stuff.

unblocked the CI stuff by the way, looks like all that stuff is failing because of this import:

```python
from python_modules.dagster.dagster._cli.job import get_config_from_args
```

which should just be `dagster._cli.job`

I'd expect that once you fix that up, you should be able to run tests locally